File: benchmark/method.py
# Bound instance-method call and return overhead.
import time

class Worker:
    def transform(self, value):
        return value * 1.000_001 + 0.5


worker = Worker()
t0 = time.perf_counter()
i = 0
value = 0.0

while i < 500_000:
    value = worker.transform(value)
    i += 1

print((time.perf_counter() - t0) * 1000)

File: benchmark/method.lua
-- Bound instance-method call and return overhead.
local Worker = {}
Worker.__index = Worker

function Worker:transform(value)
    return value * 1.000001 + 0.5
end

local worker = setmetatable({}, Worker)
local t0 = os.clock()
local i = 0
local value = 0.0

while i < 500000 do
    value = worker:transform(value)
    i = i + 1
end

print((os.clock() - t0) * 1000)
